histomicsui/web_client/vue/components/EditHeatmapOrGridData.vue

```diff
@@ -0,0 +1,229 @@
+<script>
+import tinycolor from 'tinycolor2';
+import _ from 'underscore';
+import ColorPickerInput from './ColorPickerInput.vue';
+export default {
+    props: ['elementData'],
+    emits: ['submit', 'cancel'],
+    components: {
+        ColorPickerInput
+    },
+    data() {
+        return {
+            type: this.elementData.type,
+            interpretation: this.elementData.interpretation,
+            radius: this.elementData.radius,
+            normalizeRange: this.elementData.normalizeRange,
+            colorRange: _.clone(this.elementData.colorRange),
+            colorRangeData: [],
+            rangeValues: _.clone(this.elementData.rangeValues),
+            minColor: this.elementData.minColor,
+            maxColor: this.elementData.maxColor,
+            stepped: this.elementData.stepped,
+            scaleWithZoom: this.elementData.scaleWithZoom,
+            validationErrors: []
+        };
+    },
+    computed: {
+        headerMessage() {
+            if (this.type === 'heatmap') {
+                return 'Edit Heatmap Element';
+            }
+            return 'Edit Grid Data Element';
+        }
+    },
+    methods: {
+        getColorString(color) {
+            return tinycolor(color).toString();
+        },
+        addColor(index) {
+            const newEntry = {
+                r: 0,
+                g: 0,
+                b: 0,
+                a: 0,
+                colorString: 'rgba(0, 0, 0, 0)',
+                value: 0
+            }
+            this.colorRangeData.splice(index + 1, 0, newEntry);
+            this.updateRangeDataKeys();
+        },
+        removeColor(index) {
```

Comment:
If there is only one color, don't allow its removal.

Reply:
8a1fc26